### PR TITLE
Use github actions to generate cask json

### PR DESCRIPTION
**Pull request by cmbernard333**
Issue reference: [Add Casks to formulae.brew.sh] (https://github.com/Homebrew/formulae.brew.sh/issues/13).

Generate cask json using github actions similar to [#37123](https://github.com/Homebrew/homebrew-core/pull/37123) for homebrew-core.

### CODE REVIEW

**Comment by reitermarkus** [comment]
Should this include the other cask taps?

File: .github/main.workflow [hunk 1]
```diff
@@ -60,3 +60,14 @@ action "sync_labels" {
...
+
+workflow "Trigger generate formulae.brew.sh on push" {
+  on = "push"
+  resolves = ["Generate formulae.brew.sh"]
+}
+
+action "Generate formulae.brew.sh" {
+  uses = "docker://linuxbrew/brew"
+  runs = ".github/main.workflow.sh"
+  secrets = ["ANALYTICS_JSON_KEY", "FORMULAE_DEPLOY_KEY"]
```
Comment (MikeMcQuaid) [inline .github/main.workflow]: Note to self: I'll need to add these before this will work.

File: .github/main.workflow.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+#!/bin/bash
```
Comment (MikeMcQuaid) [inline .github/main.workflow.sh]: Dupliated.

File: .github/main.workflow.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# configure git
+git config --global user.name "BrewTestBot"
+git config --global user.email "homebrew-test-bot@lists.sfconservancy.org"
+
+# create stubs so build dependencies aren't incorrectly flagged as missing
+for i in python svn unzip xz
+do
+  touch /usr/bin/$i
+  chmod +x /usr/bin/$i
+done
```
Comment (MikeMcQuaid) [inline .github/main.workflow.sh]: These shouldn't be needed here.

File: .github/main.workflow.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+# setup Homebrew/homebrew-cask instead of Linuxbrew's
```
Comment (MikeMcQuaid) [inline .github/main.workflow.sh]: This comment no longer makes sense.

File: .github/main.workflow.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+CASK_DIR="$(brew --repo homebrew/cask)"
+mkdir -p "$CASK_DIR"
+rm -rf "$CASK_DIR"
+ln -s "$PWD" "$CASK_DIR"
+
+# setup Homebrew environment
+export HOMEBREW_NO_AUTO_UPDATE=1
+export HOMEBREW_NO_ANALYTICS=1
+export HOMEBREW_FORCE_HOMEBREW_ON_LINUX=1
```
Comment (MikeMcQuaid) [inline .github/main.workflow.sh]: Are you sure this is needed for Cask generation?

Comment (cmbernard333) [reply .github/main.workflow.sh]: If we aren't doing any `install` commands we could probably remove the first one. Same with analytics.

I would expect the third one as this is being run inside a docker container.

**Comment by cmbernard333** [comment]
> Should this include the other cask taps?

I'm not sure @reitermarkus - I am intending this only to do `homebrew-cask` right now much in the same way `homebrew-core` is done.

File: .github/main.workflow.sh (hunk 2 above)
Comment (cmbernard333) [reply .github/main.workflow.sh]: Removed

File: .github/main.workflow.sh (hunk 3 above)
Comment (cmbernard333) [reply .github/main.workflow.sh]: Removed

File: .github/main.workflow.sh (hunk 4 above)
Comment (cmbernard333) [reply .github/main.workflow.sh]: Removed

**Comment by reitermarkus** [comment]
This action will be synced to the other cask taps by another action, so either make sure it works with any cask tap or make sure it isn't synced to/enabled on the other taps.

**Comment by cmbernard333** [comment]
In theory it should work for any tap. It's design just to call `to_h` on each cask within a tap and jsonify that. If they are not well formed or do not adhere to the cask DSL then it would probably throw an error.

**Comment by reitermarkus** [comment]
Analytics includes all taps, so it would make sense for this to also apply to all taps.

File: .github/main.workflow.sh [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+# silence bundler complaining about being root
+mkdir ~/.bundle
+echo 'BUNDLE_SILENCE_ROOT_WARNING: "1"' > ~/.bundle/config
+
+# configure git
+git config --global user.name "BrewTestBot"
+git config --global user.email "homebrew-test-bot@lists.sfconservancy.org"
+
+CASK_DIR="$(brew --repo homebrew/cask)"
```
Comment (reitermarkus) [inline .github/main.workflow.sh]:
```suggestion
CASK_DIR="$(brew --repo "$GITHUB_REPOSITORY")"
```

File: .github/main.workflow.sh [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+mkdir ~/.bundle
+echo 'BUNDLE_SILENCE_ROOT_WARNING: "1"' > ~/.bundle/config
+
+# configure git
+git config --global user.name "BrewTestBot"
+git config --global user.email "homebrew-test-bot@lists.sfconservancy.org"
+
+CASK_DIR="$(brew --repo "$GITHUB_REPOSITORY")"
+mkdir -p "$CASK_DIR"
+rm -rf "$CASK_DIR"
```
Comment (reitermarkus) [inline .github/main.workflow.sh]: `rm` after `mkdir` doesn't make sense.

Comment (MikeMcQuaid) [reply .github/main.workflow.sh]: It's to ensure the parent hierarchy is created

**Comment by cmbernard333** [comment]
@MikeMcQuaid what are the next steps here?

**Comment by MikeMcQuaid** [comment]
@cmbernard333 I've been on vacation. Will get to this ASAP.

File: .github/main.workflow.sh (hunk 5 above)
Comment (MikeMcQuaid) [reply .github/main.workflow.sh]: Let's try to remove them all for now (and `unset HOMEBREW_NO_ANALYTICS` below).

**Comment by MikeMcQuaid** [comment]
Let's see how this goes after merge and I'll open up a follow-up as needed. Thanks @cmbernard333!

**Comment by cmbernard333** [comment]
@MikeMcQuaid how did this work out? I'm not sure where to check the status. Is there an issue or enhancement for generating the HTML for casks? That is something I could take on in the next week or so.

**Comment by MikeMcQuaid** [comment]
@cmbernard333 Thanks for checking in! We're getting Cask JSON and analytics data going in https://github.com/Homebrew/formulae.brew.sh/tree/master/_data so, yeh, I think it's a matter of getting the HTML working, linking it (e.g. on the analytics pages and index) and then everything is done!

**Pull request merged**